src/python/WMCore/Storage/StageOutImpl.py

```diff
@@ -203,30 +202,62 @@ def __call__(self, protocol, inputPFN, targetPFN, options=None, checksums=None):
                 msg += "Error details:\n{}\n".format(str(ex))
                 logging.error(msg)
                 if retryCount == self.numRetries:
-                    #  //
+                    # //
                     # // last retry, propagate exception
                     # //
+                    logging.error("Maximum retries exhausted when trying to create the output directory")
                     raise ex
                 time.sleep(self.retryPause)
 
         # //
         # // Create the command to be used.
         # //
-        command = self.createStageOutCommand(sourcePFN, targetPFN, options, checksums)
-        #  //
+        # // This will actually only enforce the definition of the bearer token variables, still x509 auth activated as a backup
+        try:
+            command = self.createStageOutCommand(sourcePFN, targetPFN, options, checksums, authmethod="TOKEN")
```

Comment:
Same comment about the type of exception as bellow. 

In addition, it seems to me this method `self.createStageOutCommand`  could raise only `NotIMplemented` Error: https://github.com/dmwm/WMCore/pull/12218/files#diff-56bc7fd0d8d32603b392a281d22963b86c3f0d45e73c3cee08304b051d1fdc8eR143: 
```
 raise NotImplementedError("StageOutImpl.createStageOutCommand")
 ```
 Or is it supposed to be overwritten through inheritance from somewhere else later at runtime?

Reply:
As below, I've added this try-except to manage the cases where a submodule does not support `authmethod` yet, as for now we are supposed to support tokens only for gfal and xrootd. TypeError is the type of exception returned if a submodule different from GFAL2Impl or XRDImpl is used. IMU, it is because the submodule does not support the new additional argument